engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/source/FlinkMiniClusterDataGenerator.scala

```diff
@@ -0,0 +1,132 @@
+
+import java.net.URL
+import java.nio.charset.StandardCharsets
+import java.nio.file.{DirectoryStream, Files, Path}
+import scala.jdk.CollectionConverters._
+import scala.util.{Failure, Success, Try}
+
+class FlinkMiniClusterDataGenerator(flinkTableSchema: Schema) extends LazyLogging {
+
+  // TODO: check the minicluster releases memory properly and if not, refactor to reuse one minicluster per all usages
+  def generateTestData(amountOfRecordsToGenerate: Int): TestData =
+    // setting context classloader because Flink in multiple places relies on it and without this temporary override it doesnt have
+    // the necessary classes
+    ThreadUtils.withThisAsContextClassLoader(getClass.getClassLoader) {
+      val env = TableEnvironment.create(miniClusterTestingEnvConfig)
+      createGeneratorTable(env, amountOfRecordsToGenerate)
+      val (tempDirForOutputTable, tempDirUrl) = createTempDirectoryForTestResults()
+      createOutputFileTable(env, tempDirUrl)
+      val generatedRowsTry = Try {
+        insertDataAndAwait(
+          env = env,
+          inputTableName = dataGenerationInputInternalTableName,
+          outputTableName = dataGenerationOutputFileInternalTableName
+        )
+        readRecordsFromFilesUnderPath(tempDirForOutputTable)
+      }
+      tryToDeleteDirectoryWithLogging(tempDirForOutputTable)
+      val rows = generatedRowsTry.get
+      TestData(rows.map(TestRecord(_)))
+    }
+
+  private def readRecordsFromFilesUnderPath(path: Path) = {
+    val filesUnderPath: List[Path] = {
+      val dirStream: DirectoryStream[Path] = Files.newDirectoryStream(path)
+      val outputFiles                      = dirStream.asScala.toList
+      dirStream.close()
+      outputFiles
+    }
+    filesUnderPath
+      .flatMap(f => FileUtils.readLines(f.toFile, StandardCharsets.UTF_8).asScala)
+      .map(parse)
+      .sequence
+      .getOrElse(throw new IllegalStateException("Couldn't parse record from test data dump"))
+  }
+
+  private def insertDataAndAwait(env: TableEnvironment, inputTableName: String, outputTableName: String): Unit = {
+    val inputTable = env.from(inputTableName)
+    inputTable.insertInto(outputTableName).execute().await()
+  }
+
+  private def createTempDirectoryForTestResults(): (Path, URL) = {
+    val tempDir    = Files.createTempDirectory(tempTestDataOutputTablePrefix)
+    val tempDirUrl = tempDir.toUri.toURL
+    logger.debug(s"Created temporary directory for dumping test data at: '$tempDirUrl'")
+    tempDir -> tempDirUrl
+  }
+
+  private def createGeneratorTable(env: TableEnvironment, amountOfRecordsToGenerate: Int): Unit = env.createTable(
+    dataGenerationInputInternalTableName,
+    TableDescriptor
+      .forConnector("datagen")
+      .option("number-of-rows", amountOfRecordsToGenerate.toString)
+      .schema(flinkTableSchema)
+      .build()
+  )
+
+  private def createOutputFileTable(env: TableEnvironment, tempDirUrl: URL): Unit = env.createTable(
```

Comment:
maybe we should generate the temp dir inside this method and instead of returning Unit, we will return the file where the output for table will be stored?